server/api/src/rsDataObjOpen.cpp

```diff
@@ -1227,6 +1228,39 @@ namespace
             }
         }
 
+        // Updates to the atime only consider read operations.
+        // TODO reword this statement.
+        if ((dataObjInp->openFlags & O_ACCMODE) == O_RDONLY) {
+            // Enqueue an access time update if the last time the replica was accessed is earlier than
+            // the modification time or the configured amount of time has elapsed since the last update.
```

Comment:
This limits updates to atime to opens for reading only.

Linux limits atime updates to read operations - i.e. bytes are actually read from the file. This implementation only queues an atime update during the open (for reads only). I decided to not put this in the read API because I didn't want to risk flooding the queue with atime updates for the same replica. **Remember, the queue has a fixed size**. This decision also keeps us from having to develop logic for dealing with multiple calls to the read API.

So, whereas Linux likely updates the atime on each read operation, iRODS updates the atime when opening a replica for reading.

Happy to discuss alternatives.

Reply:
agreed.  this seems easier, and definitely a hidden implementation detail we can tweak later without any external effect/surface area change.

Reply:
Cool. Resolving.